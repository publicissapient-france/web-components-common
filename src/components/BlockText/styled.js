import styled from 'styled-components';
import { device } from '../../styles/constants';
import { generateSize, generatePadding, generateMargin, generateFontProperties } from '../../utils/StyleGenerator';


export const Container = styled.div.attrs(props => ({
    responsive: props.responsive,
    basis : props.basis

}))`
   width : 100%;
   height :auto;
   display : flex;
   flex-direction : column;
   align-items : center;
   justify-content  : center;   
   
   ${ props => props.responsive.map((size, i) => `
         @media ${ device[size] } {
            background-color:${ `rgba(${props.basis[size].color.rgb},${props.basis[size].opacity.value})` };
                   
            ${ props.basis ?  generateSize(props.basis, size) : '' }        
            ${ props.basis ?  generatePadding(props.basis, size) : '' }        
            ${ props.basis ?  generateMargin(props.basis, size) : '' }        
             
         }`)
}; 
`;

export const Text = styled.p.attrs(props => ({
    responsive: props.responsive,
    typography : props.typography,
    basis : props.basis
}))`
   ${ props => props.responsive.map(size => `
         @media ${ device[size] } {
            color:${ `rgba(${props.typography[size].color.rgb},${props.typography[size].opacity.value})` };
           
            ${ props.typography ?  generateFontProperties(props.typography, size) : '' }
            ${ props.basis ?  generatePadding(props.basis, size) : '' }        

         }`)
};
`;

export const Content = styled.div.attrs(props => ({
    responsive: props.responsive,
    typography : props.typography,
    basis : props.basis
}))`

    width : 100%;
    word-break: break-word;
    
   ${ props => props.responsive.map(size => `
         @media ${ device[size] } {
            color:${ `rgba(${props.typography[size].color.rgb},${props.typography[size].opacity.value})` };
            
            ${ props.typography ?  generateFontProperties(props.typography, size) : '' }
            ${ props.basis ?  generatePadding(props.basis, size) : '' } 
          
            & a {
               color:${ `rgba(${props.typography[size].color.rgb},${props.typography[size].opacity.value})` };
            }
            
         }`)
    };
`;