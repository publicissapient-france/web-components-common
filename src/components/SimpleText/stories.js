import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, text, number } from "@storybook/addon-knobs/react";
import SimpleText from "./index";

storiesOf("Text", module)
    .addDecorator(withKnobs)
    .add("default", () => <SimpleText />)
    .add("Text and Input with Knobs", () => {
        const name = text("Name", "React App");
        const level = number("Level", 3);

        const content = `This is a ${name} and it has Level ${level} .`;

        return <SimpleText content={content} />;
    })
    .add("Input with Knobs", () => {
        const input = text("Input", "Type away ...");

        return <SimpleText content={input} />;
    });