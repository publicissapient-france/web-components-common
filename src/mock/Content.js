
//const htmlFR = <p>Cette année, Frenchkit fêtera ses quatre ans ! Nous avons adoré organiser les trois premières éditions. Nous avons expérimenté beaucoup de choses ; certaines étaient géniales… d’autres moins. En 2019, nous continuerons de faire de Frenchkit une conférence vivante en prenant ce que vous avez le plus aimé et en introduisant de nouvelles idées là où nous pensons qu’il y a place à l’amélioration :</p><ul> <li>Un nouveau lieu formidable.</li> <li>Des discussions plus expertes, plus axées sur le code et plus pratiques!</li> <li>Le retour des salles de classe que vous avez aimées en 2017.</li> <li>Plus de temps et d’activités communautaires.</li></ul><p>Nous vous tiendrons régulièrement au courant de l’avancement de l’événement et de notre programme.</p>);
//const htmlEN = <p>Cette année, Frenchkit fêtera ses quatre ans ! Nous avons adoré organiser les trois premières éditions. Nous avons expérimenté beaucoup de choses ; certaines étaient géniales… d’autres moins. En 2019, nous continuerons de faire de Frenchkit une conférence vivante en prenant ce que vous avez le plus aimé et en introduisant de nouvelles idées là où nous pensons qu’il y a place à l’amélioration :</p><ul> <li>Un nouveau lieu formidable.</li> <li>Des discussions plus expertes, plus axées sur le code et plus pratiques!</li> <li>Le retour des salles de classe que vous avez aimées en 2017.</li> <li>Plus de temps et d’activités communautaires.</li></ul><p>Nous vous tiendrons régulièrement au courant de l’avancement de l’événement et de notre programme.</p>);

export const Content = {
    content: {
        html: {
            0: '<p>Contenu</p',
            1: '<p>Inner</p>'
        }
    },
    responsiveSettings: ['A'],
    settings: {
        color: {
            A: {
                hex: '#000000',
                name: 'black',
                rgb: '0,0,0',
                shade: ''
            }
        },
        font: {
            A: {
                family: 'Arial',
                letterSpacing: '0',
                lineHeight: '24',
                size: '16',
                style: null,
                theme: 'Paragraoh',
                typeface: 'sans-serif',
                weight: ['Regular', 400]
            }
        },
        text: {
            A: {
                align: 'center',
                decoration: null,
                transform: null
            }
        },
        opacity: {
            A: {
                value: '0.95'
            }
        },
        seo: {
            tag: 'h2'
        }

    }
};

