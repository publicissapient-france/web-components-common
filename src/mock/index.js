export * from './Template';
export * from './Text';
export * from './Image';
export * from './Images';