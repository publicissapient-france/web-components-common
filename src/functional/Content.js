import React from 'react';
import PropTypes from 'prop-types';
import {ContentCommon} from "../styles/common.styled";
import {getContentProps} from "../utils/gettersProperties";


const Content = ({field, language}) => {
    const html = field.content.html ? field.content.html[language] : null
    if (!html) return null;
    return (
        <ContentCommon {...getContentProps(field)} dangerouslySetInnerHTML={{__html: html}}/>
    );
};

Content.defaultProps = {}

export default Content;
