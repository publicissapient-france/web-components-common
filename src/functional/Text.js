import React from 'react';
import PropTypes from 'prop-types';
import {TextCommon} from "../styles/common.styled";
import {getTextProps} from "../utils/gettersProperties";

const Text = ({field, language}) => {
    const content = field.content.text ? field.content.text[language] : null
    if (!content) return null;
    return (<TextCommon {...getTextProps(field)}>{content}</TextCommon>);
};

Text.propTypes = {
    field : PropTypes.object,
    language : PropTypes.number
};

export default Text;
