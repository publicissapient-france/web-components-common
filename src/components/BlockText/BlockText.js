import React from 'react';
import {Container} from './styled';
import Text from '../../functional/Text'
import Content from '../../functional/Content'
import PropTypes from 'prop-types';
import {getTemplateProps} from "../../utils/gettersProperties";

const buildComponent = (fields, field,language, key) => {
    if (!fields[field]) return
    switch (field) {
        case 'Title':
            return <Text key={key} field={fields[field]} language={language}/>;

        case 'Tagline':
            return <Text key={key} field={fields[field]} language={language}/>;

        case 'Content':
            return <Content key={key} field={fields[field]} language={language}/>;

        default :
            return null;
    }
}

const BlockText = ({fields, order, language}) => {
    return (<Container {...getTemplateProps(fields.Template)}>
        {
            order ? order.map((fieldName, i) => buildComponent(fields, fieldName,language, i))
                : ['Title', 'Tagline', 'Content'].map((fieldName, i) => buildComponent(fields, fieldName,language, i))
        }
    </Container>)
}

BlockText.defaultProps = {};

BlockText.propTypes = {
    order: PropTypes.arrayOf(PropTypes.string),
    fields: PropTypes.shape({
        Template: PropTypes.object,
        Title: PropTypes.object,
        Tagline: PropTypes.object,
        Content: PropTypes.object
    }),
    language: PropTypes.number.isRequired
};

export default BlockText;
