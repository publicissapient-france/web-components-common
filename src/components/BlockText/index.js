import BlockText from './BlockText';
export default BlockText;
