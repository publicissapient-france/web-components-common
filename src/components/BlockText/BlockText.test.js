import React from 'react';
import { shallow } from 'enzyme';
import BlockText from './BlockText';
import {Template, TextMock, ContentMock} from "../../../stories/mock";
import Text from '../../functional/Text';
import Content from '../../functional/Content';

const fieldsMock =  {
    Template: {
        content: {},
        responsiveSettings: ['A'],
        settings : Template
    },
    Title: TextMock,
    Tagline: TextMock,
    Content : ContentMock
};

describe('BlockText', () => {
    it('should render correctly', () => {
        const component = shallow(<BlockText fields={fieldsMock} language={0}/>);

        expect(component).toMatchSnapshot();
    });
});

describe('BlockText', () => {
    it('should render correctly title with order prop', () => {
        const component = shallow(<BlockText order={['Title']} fields={fieldsMock} language={0}/>);

        expect(component.find(Text)).toHaveLength(1);
    });
});

describe('BlockText', () => {
    it('should render correctly Title and Tagline with order prop', () => {
        const component = shallow(<BlockText order={['Title', 'Tagline']} fields={fieldsMock} language={0}/>);

        expect(component.find(Text)).toHaveLength(2);
    });
});

describe('BlockText', () => {
    it('should render correctly Title,Tagline, Content  and respect order prop', () => {
        const component = shallow(<BlockText order={['Title', 'Tagline', 'Content']} fields={fieldsMock} language={0}/>);

        expect(component.find(Text)).toHaveLength(2);
        expect(component.find(Content)).toHaveLength(1);
    });
});

describe('BlockText', () => {
    it('should render correctly order of children', () => {
        const component = shallow(<BlockText order={['Title', 'Content', 'Tagline']} fields={fieldsMock} language={0}/>);
        expect(component.childAt(0).is(Text)).toBeTruthy()
        expect(component.childAt(1).is(Content)).toBeTruthy()
        expect(component.childAt(2).is(Text)).toBeTruthy()
    });
});

describe('BlockText', () => {
    it('should render only element declare on props order', () => {
        const component = shallow(<BlockText order={['Title']} fields={fieldsMock} language={0}/>);
        expect(component.children()).toHaveLength(1);
    });
});

describe('BlockText', () => {
    it('should render all elements without  props order', () => {
        const component = shallow(<BlockText fields={fieldsMock} language={0}/>);
        expect(component.children()).toHaveLength(3);
    });
});