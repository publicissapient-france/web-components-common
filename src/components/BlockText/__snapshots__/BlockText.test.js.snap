// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockText should render correctly 1`] = `
<styled__Container
  border={null}
  responsive={
    Array [
      "A",
    ]
  }
>
  <Text
    field={
      Object {
        "content": Object {
          "text": Object {
            "0": "Un text",
            "1": "Some text",
          },
        },
        "responsiveSettings": Array [
          "M",
          "T",
          "D",
        ],
        "settings": Object {
          "basis": Object {
            "D": Object {
              "padding": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "M": Object {
              "padding": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "T": Object {
              "padding": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
          },
          "border": Object {
            "D": Object {
              "color": Object {
                "hex": "transparent",
                "name": "Transparent",
                "rgb": "",
                "shade": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "radius": Object {
                "bottomLeft": "0",
                "bottomRight": "0",
                "topLeft": "0",
                "topRight": "0",
              },
              "width": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "M": Object {
              "color": Object {
                "hex": "transparent",
                "name": "Transparent",
                "rgb": "",
                "shade": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "radius": Object {
                "bottomLeft": "0",
                "bottomRight": "0",
                "topLeft": "0",
                "topRight": "0",
              },
              "width": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "T": Object {
              "color": Object {
                "hex": "transparent",
                "name": "Transparent",
                "rgb": "",
                "shade": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "radius": Object {
                "bottomLeft": "0",
                "bottomRight": "0",
                "topLeft": "0",
                "topRight": "0",
              },
              "width": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
          },
          "seo": Object {
            "tag": "h2",
          },
          "typography": Object {
            "D": Object {
              "color": Object {
                "hex": "#000000",
                "name": "black",
                "rgb": "0,0,0",
                "shade": null,
              },
              "font": Object {
                "family": null,
                "letterSpacing": "0",
                "lineHeight": null,
                "size": null,
                "style": null,
                "theme": "Title3",
                "typeface": null,
                "weight": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "text": Object {
                "align": "left",
                "decoration": null,
                "transform": null,
              },
            },
            "M": Object {
              "color": Object {
                "hex": "#000000",
                "name": "black",
                "rgb": "0,0,0",
                "shade": null,
              },
              "font": Object {
                "family": null,
                "letterSpacing": "0",
                "lineHeight": null,
                "size": null,
                "style": null,
                "theme": "Title3",
                "typeface": null,
                "weight": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "text": Object {
                "align": "left",
                "decoration": null,
                "transform": null,
              },
            },
            "T": Object {
              "color": Object {
                "hex": "#000000",
                "name": "black",
                "rgb": "0,0,0",
                "shade": null,
              },
              "font": Object {
                "family": null,
                "letterSpacing": "0",
                "lineHeight": null,
                "size": null,
                "style": null,
                "theme": "Title3",
                "typeface": null,
                "weight": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "text": Object {
                "align": "left",
                "decoration": null,
                "transform": null,
              },
            },
          },
        },
      }
    }
    key="0"
    language={0}
  />
  <Text
    field={
      Object {
        "content": Object {
          "text": Object {
            "0": "Un text",
            "1": "Some text",
          },
        },
        "responsiveSettings": Array [
          "M",
          "T",
          "D",
        ],
        "settings": Object {
          "basis": Object {
            "D": Object {
              "padding": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "M": Object {
              "padding": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "T": Object {
              "padding": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
          },
          "border": Object {
            "D": Object {
              "color": Object {
                "hex": "transparent",
                "name": "Transparent",
                "rgb": "",
                "shade": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "radius": Object {
                "bottomLeft": "0",
                "bottomRight": "0",
                "topLeft": "0",
                "topRight": "0",
              },
              "width": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "M": Object {
              "color": Object {
                "hex": "transparent",
                "name": "Transparent",
                "rgb": "",
                "shade": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "radius": Object {
                "bottomLeft": "0",
                "bottomRight": "0",
                "topLeft": "0",
                "topRight": "0",
              },
              "width": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "T": Object {
              "color": Object {
                "hex": "transparent",
                "name": "Transparent",
                "rgb": "",
                "shade": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "radius": Object {
                "bottomLeft": "0",
                "bottomRight": "0",
                "topLeft": "0",
                "topRight": "0",
              },
              "width": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
          },
          "seo": Object {
            "tag": "h2",
          },
          "typography": Object {
            "D": Object {
              "color": Object {
                "hex": "#000000",
                "name": "black",
                "rgb": "0,0,0",
                "shade": null,
              },
              "font": Object {
                "family": null,
                "letterSpacing": "0",
                "lineHeight": null,
                "size": null,
                "style": null,
                "theme": "Title3",
                "typeface": null,
                "weight": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "text": Object {
                "align": "left",
                "decoration": null,
                "transform": null,
              },
            },
            "M": Object {
              "color": Object {
                "hex": "#000000",
                "name": "black",
                "rgb": "0,0,0",
                "shade": null,
              },
              "font": Object {
                "family": null,
                "letterSpacing": "0",
                "lineHeight": null,
                "size": null,
                "style": null,
                "theme": "Title3",
                "typeface": null,
                "weight": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "text": Object {
                "align": "left",
                "decoration": null,
                "transform": null,
              },
            },
            "T": Object {
              "color": Object {
                "hex": "#000000",
                "name": "black",
                "rgb": "0,0,0",
                "shade": null,
              },
              "font": Object {
                "family": null,
                "letterSpacing": "0",
                "lineHeight": null,
                "size": null,
                "style": null,
                "theme": "Title3",
                "typeface": null,
                "weight": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "text": Object {
                "align": "left",
                "decoration": null,
                "transform": null,
              },
            },
          },
        },
      }
    }
    key="1"
    language={0}
  />
  <Content
    field={
      Object {
        "content": Object {
          "html": Object {
            "0": "<p>Contenu</p",
            "1": "<p>Inner</p>",
          },
        },
        "responsiveSettings": Array [
          "A",
        ],
        "settings": Object {
          "basis": Object {
            "D": Object {
              "padding": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "M": Object {
              "padding": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "T": Object {
              "padding": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
          },
          "border": Object {
            "D": Object {
              "color": Object {
                "hex": "transparent",
                "name": "Transparent",
                "rgb": "",
                "shade": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "radius": Object {
                "bottomLeft": "0",
                "bottomRight": "0",
                "topLeft": "0",
                "topRight": "0",
              },
              "width": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "M": Object {
              "color": Object {
                "hex": "transparent",
                "name": "Transparent",
                "rgb": "",
                "shade": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "radius": Object {
                "bottomLeft": "0",
                "bottomRight": "0",
                "topLeft": "0",
                "topRight": "0",
              },
              "width": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
            "T": Object {
              "color": Object {
                "hex": "transparent",
                "name": "Transparent",
                "rgb": "",
                "shade": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "radius": Object {
                "bottomLeft": "0",
                "bottomRight": "0",
                "topLeft": "0",
                "topRight": "0",
              },
              "width": Object {
                "bottom": "0",
                "left": "0",
                "right": "0",
                "top": "0",
              },
            },
          },
          "typography": Object {
            "D": Object {
              "color": Object {
                "hex": "#000000",
                "name": "black",
                "rgb": "0,0,0",
                "shade": null,
              },
              "font": Object {
                "family": null,
                "letterSpacing": "0",
                "lineHeight": null,
                "size": null,
                "style": null,
                "theme": "Paragraph",
                "typeface": null,
                "weight": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "text": Object {
                "align": "left",
                "decoration": null,
                "transform": null,
              },
            },
            "M": Object {
              "color": Object {
                "hex": "#000000",
                "name": "black",
                "rgb": "0,0,0",
                "shade": null,
              },
              "font": Object {
                "family": null,
                "letterSpacing": "0",
                "lineHeight": null,
                "size": null,
                "style": null,
                "theme": "Paragraph",
                "typeface": null,
                "weight": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "text": Object {
                "align": "left",
                "decoration": null,
                "transform": null,
              },
            },
            "T": Object {
              "color": Object {
                "hex": "#000000",
                "name": "black",
                "rgb": "0,0,0",
                "shade": null,
              },
              "font": Object {
                "family": null,
                "letterSpacing": "0",
                "lineHeight": null,
                "size": null,
                "style": null,
                "theme": "Paragraph",
                "typeface": null,
                "weight": null,
              },
              "opacity": Object {
                "value": "1",
              },
              "text": Object {
                "align": "left",
                "decoration": null,
                "transform": null,
              },
            },
          },
        },
      }
    }
    key="2"
    language={0}
  />
</styled__Container>
`;
